dask/dataframe/tests/test_utils_dataframe.py

```diff
@@ -521,3 +523,30 @@ def test_assert_eq_sorts():
     assert_eq(df1, df2_r, check_index=False)
     with pytest.raises(AssertionError):
         assert_eq(df1, df2_r)
+
+
+def test_assert_eq_scheduler():
+    using_custom_scheduler = False
+
+    def custom_scheduler(*args, **kwargs):
+        nonlocal using_custom_scheduler
+        try:
+            using_custom_scheduler = True
+            return get_sync(*args, **kwargs)
+        finally:
+            using_custom_scheduler = False
+
+    def check_custom_scheduler(part: pd.DataFrame) -> pd.DataFrame:
+        assert using_custom_scheduler
+        return part + 1
+
+    df = pd.DataFrame({"x": [1, 2, 3, 4]})
+    ddf = dd.from_pandas(df, npartitions=2)
+    ddf2 = ddf.map_partitions(check_custom_scheduler, meta=ddf)
+
+    with pytest.raises(AssertionError):
```

Comment:
Let's raise either a custom exception class (`class MyError(Exception): pass`) or raise an error with a unique message and check for that message. Otherwise it's not completely clear that this `AssertionError` is the same one coming from `check_custom_scheduler` -- it probably is, but _could_ be coming from somewhere else